polars/polars-io/src/csv/write_impl.rs

```diff
@@ -293,11 +323,18 @@ pub(crate) fn write<W: Write>(
                 for (i, col) in &mut col_iters.iter_mut().enumerate() {
                     let datetime_format = match &options.datetime_format {
                         Some(datetime_format) => Some(datetime_format.as_str()),
-                        None => unsafe { *formats.as_ref().unwrap().get_unchecked(i) },
+                        None => unsafe { *datetime_formats.get_unchecked(i) },
                     };
+                    let time_zone = unsafe { *time_zones.get_unchecked(i) };
```

Comment:
Can we pass the `time_zones` slice to `write_anyvalue`. Most columns don't need to access the time_zone, so we don't need to take the value unless we write it.

If we also send the current index `i` to `write_anyvalue`. That function could determine itself if it should get the `time_zone` or not.